activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -11,8 +11,8 @@ def explain(arel, binds = [], options = [])
         end
 
         # Queries the database and returns the results in an Array-like object
-        def query(sql, name = nil) # :nodoc:
-          result = internal_execute(sql, name)
+        def query(sql, name = nil, allow_retry: true, materialize_transactions: true) # :nodoc:
```

Comment:
```suggestion
        def query(sql, name = nil, allow_retry: false, materialize_transactions: true) # :nodoc:
```

As mentioned, retrying something that shouldn't is a really bad bug. Let's default to the safe setting.

Reply:
Is there something different between this (postgresql) `query` than the abstract one? Both should only be used internally by SCHEMA queries. This change makes them have the same signature (abstract `query` was changed in https://github.com/rails/rails/commit/b4c19d038e558d12c3e13317759ca13587c41a16).

Reply:
Actually, this change should probably be its own PR, might be worth backporting.

Reply:
https://github.com/rails/rails/pull/54439